### PR TITLE
feat: add USB device list/watch

### PR DESCRIPTION
**Pull request by eventualbuddha**
Adds `kiosk.getDeviceList` and `kiosk.onDeviceUpdate` to get the current device list and watch for updates.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -5,6 +5,11 @@ import getURL from './utils/getURL'
 import registerPrintHandler from './ipc/print'
 import registerGetBatteryInfoHandler from './ipc/get-battery-info'
 import registerGetPrinterInfoHandler from './ipc/get-printer-info'
+import registerGetDeviceListHandler from './ipc/get-device-list'
+import registerSubscribeToDevices from './ipc/subscribe-to-devices'
+
+// usbDetection.startMonitoring()
```
Comment (benadida) [inline src/index.ts]: remove commented out code

File: src/ipc/subscribe-to-devices.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+/**
+ * Subscribe to add/remove USB device events.
+ */
+export default function register(ipcMain: IpcMain): void {
+  ipcMain.handle(channel, (event: IpcMainInvokeEvent) => {
+    const webContents = event.sender
+
+    usbDetection.startMonitoring()
```
Comment (benadida) [inline src/ipc/subscribe-to-devices.ts]: pedantic, but does it matter if you `startMonitoring` before you add event handlers? I know, JS event loop etc, just wondering if it is easier to read if you register event handlers first.

File: src/preload.ts [hunk 3]
```diff
@@ -8,6 +8,45 @@ import {
...
 } from './ipc/get-printer-info'
+import { channel as getDeviceListChannel } from './ipc/get-device-list'
+import { Device } from 'usb-detection'
+import {
+  channel as subscribeToDevicesChannel,
+  UpdateType,
+  deviceUpdateChannel,
+} from './ipc/subscribe-to-devices'
+
+class Listeners<A extends any[], C extends Function = (...args: A) => void> {
```
Comment (benadida) [inline src/preload.ts]: feels a bit like overkill to build in ability for multiple listeners? I can live with it, just wondering if this complexity is needed.

File: src/ipc/subscribe-to-devices.ts (hunk 2 above)
Comment (eventualbuddha) [reply src/ipc/subscribe-to-devices.ts]: FWIW that's what the [README example does](https://github.com/MadLittleMods/node-usb-detection).

File: src/index.ts (hunk 1 above)
Comment (eventualbuddha) [reply src/index.ts]: 👍

File: src/ipc/subscribe-to-devices.ts (hunk 2 above)
Comment (benadida) [reply src/ipc/subscribe-to-devices.ts]: Fair enough!

File: src/preload.ts (hunk 3 above)
Comment (eventualbuddha) [reply src/preload.ts]: Okay, changed it to a single listener. Note that I made it more robust in terms of actually turning monitoring off when nobody is listening, so it's already got a fair amount of inherent complexity.

File: src/ipc/subscribe-to-devices.ts (hunk 2 above)
Comment (eventualbuddha) [reply src/ipc/subscribe-to-devices.ts]: I switched it anyway. Works either way because it's actually an `EventEmitter`.

**Pull request merged**